### ISSUE TITLE
Start/Pause Filtering Feature

### ISSUE THREAD

**Issue by men1n2** [safing/portmaster#475]
**What would you like to add or change?**:

I want a Start/Pause Filtering feature to be added. A simple button or menu entry that pauses portmaster's filtering when needed and starts it.

**Why do you and others need this?**:

Sometimes, I find myself forced to shutdown portmaster because I needed some request to go through, that is blocked. For example, I use vagrant to launch different VMs throughout the day, and some VMs use NFS, which is blocked by default by portmaster. Other times, ping is blocked. etc ...
I want to be able to stop all portmaster filetring until I finish what I'm doing, and then start it easily (preferably from the tray menu).
I know that I can write the necessary rules, but it is not always as easy or straightforward as it sounds. In addition, when I'm doing some work, I don't necessarily have the time to do that.

**Comment by dhaavi** [comment]
Hey @men1n2, I think you're looking for the [Privacy Filter Module](https://docs.safing.io/portmaster/settings#filter/enable) setting.
Please note the documentation.

**Comment by men1n2** [comment]
Hello @dhaavi 

Thank you for your reply. Indeed, that is the functionality I was looking for: disabling the Privacy Filter Module.
![image](https://user-images.githubusercontent.com/5360222/146456208-8ef3b2c8-566c-417e-8dc0-baab844dfd90.png)
But I had to set the UI mode to Developer mode, and needed to go to the settings page each time I want to activate/deactivate it.

I would argue that it is a functionality that needs to be exposed to all users, not only developers, with quick toggle access. As some casual users will need to stop Portmaster filtering sometimes (but not all the time), if they encounter a website that is broken due to some blocked requests, and they need it to just work.

I would suggest a tray menu item, which would be very practical. Also, a pause/start button on the main interface.

Looking forward to hearing your opinion on adding this to the standard UI.

**Comment by dhaavi** [comment]
> I would argue that it is a functionality that needs to be exposed to all users, not only developers, with quick toggle access.

We have an update for the monitor/app view in the pipeline that will bring quick toggles. I will discuss with the team if we should put a "privacy filter" toggle there.

> As some casual users will need to stop Portmaster filtering sometimes (but not all the time), if they encounter a website that is broken due to some blocked requests, and they need it to just work.

We'd much rather improve the process of making applications work instead of just ditching the protection altogether for some time.

> I would suggest a tray menu item, which would be very practical.

We do have a tray icon, does it not show up for you? If it doesn't please create a separate issue for this.

> Also, a pause/start button on the main interface.

I feel this would encourage harmful behavior - switching the Portmaster off way too often.

**Comment by davegson** [comment]
> I would argue that it is a functionality that needs to be exposed to all users, not only developers, with quick toggle access.

I disagree here, since I do feel there is a deeper, underlying issue to be tackled. Quoting from a [reddit discussion on the same topic](https://old.reddit.com/r/safing/comments/rjeupg/is_there_a_way_to_allow_everything_for_a_certain/):

> It is possible to disable the Privacy Filter Module. But overall _we do not recommend this_, which is why it is hidden away in the Developer Interface.
> 
> Almost always there is a better way, like setting your [Default Network Action](https://docs.safing.io/portmaster/settings#filter/defaultAction) or  creating [Incoming Rules](https://docs.safing.io/portmaster/settings#filter/serviceEndpoints), where even lax rules are better than completely turning off the privacy filter.
>
> ---
>
> I feel one of the bigger issues we have at the moment is that it is not clearly communicated that incoming connections are blocked by default. Sure, we mention that in the docs, but user expectation would be for it to work.
>
> How to best communicate or tackle this issue is still open for debate. Happy to hear input and ideas.
>
> ---
>
> And more generally speaking, we know the User Interface often does not meet user expectations and are working on improving things on that front. As a teaser, simpler on/off toggles are coming to the UI which should help newcomers a lot.

**Comment by e-t-l** [comment]
@davegson @dhaavi I would argue to reconsider this issue. Contrary to creating filter rules, I would find a a filter pause (either globally or per-app) is more useful, and more privacy-preserving, if a user wants to briefly access a site that's blocked by some filter list. We the users do not want to *permanently* whitelist the domain, because the use-case it to connect to it just for one time. (maybe there could even be an XX-minute long timer to automatically unpause)
